### PR TITLE
EZP-25769: Dashboard mycontent block

### PR DESCRIPTION
**Pull request by StephaneDiot**
JIRA: https://jira.ez.no/browse/EZP-25769

## Description

This is the dashboard "My content" block which gives the content made or modified by the logged on user.

## Tests

unit and manually tested

### CODE REVIEW

**Comment by StephaneDiot** [comment]
ping @dpobel  @yannickroger

File: Resources/public/js/views/dashboard/ez-dashboardblockmycontentview.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+         */
+        _fireLoadDataEvent: function () {
+            var user = this.get('currentUser');
+
+            this.fire('locationSearch', {
+                viewName: 'my-content-' + user.get('userId'),
+                resultAttribute: 'items',
+                loadContentType: true,
+                search: {
+                    criteria: {UserMetadataCriterion: {
```
Comment (dpobel) [inline Resources/public/js/views/dashboard/ez-dashboardblockmycontentview.js]: nitpick: CS

File: Resources/public/css/theme/views/dashboard/mycontent.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+.ez-view-dashboardblockmycontentview .ez-block-option-view:before {
+    content: '\E607';
```
Comment (dpobel) [inline Resources/public/css/theme/views/dashboard/mycontent.css]: actually, this should be defined in `dashboard/asynchronous.css`

Comment (dpobel) [inline Resources/public/css/theme/views/dashboard/mycontent.css]: btw this also means you should probably remove the stylesheet in `bundles/ezplatformui/css/theme/views/dashboard/allcontent.css`

File: Resources/public/js/models/ez-usermodel.js [hunk 3]
```diff
@@ -174,7 +174,18 @@ YUI.add('ez-usermodel', function (Y) {
...
                 value: null
-            }
+            },
+
+            /**
+             * The user id
+             *
+             * @attribute userId
+             * @default null
+             * @type object
```
Comment (dpobel) [inline Resources/public/js/models/ez-usermodel.js]: `Number`

File: Resources/public/js/models/ez-usermodel.js [hunk 4]
```diff
@@ -174,7 +174,18 @@ YUI.add('ez-usermodel', function (Y) {
...
             avatar: {
                 value: null
-            }
+            },
+
+            /**
+             * The user id
+             *
+             * @attribute userId
+             * @default null
```
Comment (dpobel) [inline Resources/public/js/models/ez-usermodel.js]: nope :)

File: Tests/js/views/services/assets/ez-dashboardblocksviewservice-tests.js [hunk 5]
```diff
@@ -168,9 +168,17 @@ YUI.add('ez-dashboardblocksviewservice-tests', function (Y) {
 
         setUp: function () {
             this.rootLocationModel = new Y.Mock();
+            this.app = new Y.Mock();
+            this.user = new Y.Model();
 
+            Y.Mock.expect(this.app, {
+                method: 'get',
+                args: ['user'],
+                returns: this.rootLocationModel
```
Comment (dpobel) [inline Tests/js/views/services/assets/ez-dashboardblocksviewservice-tests.js]: `get('user')` returns the root Location Model ?

also, you need to add a test to check that `getViewParameters` returns this user.

File: Tests/js/views/dashboard/assets/ez-dashboardblockmycontentview-tests.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+                    event.loadContentType,
+                    'The loadContentType flag should be set'
+                );
+                Assert.areSame(
+                    'items', event.resultAttribute,
+                    'The search result should be set in the items attribute'
+                );
+                Assert.areSame(
+                    view.get('currentUser').get('userId'),
+                    event.search.criteria.UserMetadataCriterion.Value,
```
Comment (dpobel) [inline Tests/js/views/dashboard/assets/ez-dashboardblockmycontentview-tests.js]: missing an assert on `Target` as well

File: Resources/public/css/theme/views/dashboard/asynchronous.css [hunk 7]
```diff
@@ -51,3 +51,7 @@
 
 @media (min-width: 1280px) {
 }
+
```
Comment (dpobel) [inline Resources/public/css/theme/views/dashboard/asynchronous.css]: while at it, you can remove the empty @media just above

**Comment by dpobel** [comment]
besides the inline comment, +1

**Comment by StephaneDiot** [comment]
ping @yannickroger :pray:

**Comment by andrerom** [comment]
Nice, +1 :)

File: Tests/js/views/dashboard/ez-dashboardblockmycontentview.html [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        </tbody>
+    </table>
+</script>
+
+<script type="text/javascript" src="../../../../Resources/public/vendors/yui3/build/yui/yui.js"></script>
+<script type="text/javascript" src="./assets/ez-dashboardblockasynchronousview-tests.js"></script>
+<script type="text/javascript" src="./assets/ez-dashboardblockmycontentview-tests.js"></script>
+<script>
+    var filter = (window.location.search.match(/[?&]filter=([^&]+)/) || [])[1] || 'raw',
+            loaderFilter = (filter !== 'coverage') ? filter : {
```
Comment (dpobel) [inline Tests/js/views/dashboard/ez-dashboardblockmycontentview.html]: nitpick small cs issue

**Pull request merged**